[quant][bug] Histogram observer bug fix with min == max

supriyar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#40310 [quant][bug] Histogram observer bug fix with min == max**

Summary:

Test Plan:
python test/test_quantization.py test_histogram_observer_same_inputs

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22145908](https://our.internmc.facebook.com/intern/diff/D22145908)

supriyar committed:
[quant][bug] Histogram observer bug fix with min == max
Summary:

Test Plan:
python test/test_quantization.py test_histogram_observer_same_inputs

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar committed:
Update on "[quant][bug] Histogram observer bug fix with min == max"


Summary:

Test Plan:
python test/test_quantization.py test_histogram_observer_same_inputs

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

jerryzh168 commented on torch/quantization/observer.py:
```diff
@@ -828,10 +828,10 @@ def forward(self, x_orig):
         min_val = self.min_val
         max_val = self.max_val
-        prev_zeros = False
+        equal_values = False
```
nit: maybe `same_values`

supriyar committed:
Update on "[quant][bug] Histogram observer bug fix with min == max"


Summary:

Test Plan:
python test/test_quantization.py test_histogram_observer_same_inputs

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]
